### PR TITLE
Hdskins rewrites 2

### PR DESCRIPTION
**Pull request by Sollace**
New PR with a much cleaner history. I really don't think these can be split up because they are linearly dependent on each other

### CODE REVIEW

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java [hunk 1]
```diff
@@ -34,20 +79,52 @@ public NetClient putFile(String key, String contentType, URI file) {
...
         this.headers = headers;
 
         return this;
     }
 
-    public MoreHttpResponses send() throws IOException {
+    /**
+     * Commits and sends the request.
+     */
+    private void send() throws IOException {
```
Comment (killjoy1221) [inline src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java]: You should keep the response separated from the actual client. If you want to keep a reference to the return value, then just save it in a variable.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java [hunk 2]
```diff
@@ -28,13 +28,11 @@
             "http://skins.voxelmodpack.com",
             "http://skinmanager.voxelmodpack.com"));
 
-    MinecraftTexturesPayload loadProfileData(GameProfile profile) throws IOException;
+    Map<MinecraftProfileTexture.Type, MinecraftProfileTexture> getProfileTextures(GameProfile profile);
```
Comment (killjoy1221) [inline src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java]: Don't change this.

File: src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java [hunk 3]
```diff
@@ -87,12 +86,14 @@
     private List<ISkinModifier> skinModifiers = Lists.newArrayList();
 
     private SkinResourceManager resources = new SkinResourceManager();
-    //    private ExecutorService executor = Executors.newCachedThreadPool();
 
     private Class<? extends GuiSkins> skinsClass = null;
 
-    private HDSkinManager() {
+    public static PreviewTextureManager getPreviewTextureManager(GameProfile profile) {
```
Comment (killjoy1221) [inline src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java]: Why is this moved up here? Also, you're undoing a bunch of my work.

Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java]: A bunch of your work here was directly incompatible, and quite frankly unneeded. Preview textures are already loaded with a completable future, it's just not done by wrapping the _entire PreviewTextureManager_.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java (hunk 2 above)
Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java]: Why not? It simplifies the rest of those code and the only thing that benefits from there being a separate method is LegacySkinServer, which will already override this function to use its gateway.

File: src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java (hunk 3 above)
Comment (killjoy1221) [reply src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java]: Valhalla needs to do a network call in order to get the locations of the skins. It's also smart to use a future as soon as possible. This is all so the client thread won't lock up.

Comment (killjoy1221) [reply src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java]: If you're going to change it back, the least you can do is move it back to where it was.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java (hunk 1 above)
Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java]: NetClient is intended to wrap the request/response process as a whole. Granted, I suppose it doesn't matter now that you've gutted it.

File: src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java (hunk 3 above)
Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java]: I intentionally try to keep the static members near the top of a class where they're not going to get lost.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java (hunk 1 above)
Comment (killjoy1221) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java]: If you're familiar with requests libraries in python and javascript, this is more or less the way they operate. Call a function with the method, address, headers, and payload, get a response object back. The object usually has other functions to get the content as text, json, bytes, etc.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java (hunk 2 above)
Comment (killjoy1221) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java]: Legacy has to do several HEAD requests (it does GET, but it can get away with just HEAD) to check for a 404. the method makes that easier. Everything else can do that with a single request (I assume for Bethlehem).

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java (hunk 1 above)
Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java]: I also don't really want to get into the situation where I have to dispose of the client, and then dispose the request. It seems so much cleaner just to have one object.

Plus, previously NetClient was supposed to lazy load the request. So you could just do `client.getResponseCode()` without going through any intermediaries.

Comment (killjoy1221) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java]: You added a method that resets the client, so I assume you weren't going to get rid of it anytime soon.

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java (hunk 2 above)
Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/SkinServer.java]: That will all happen inside getProfileTextures(), then, which happens either in PreviewTextureManager (which uses a future) or in your cache builder which I can only assume is asynchronous anyway.

The outside world doesn't really need anything more than just a map of textures. (all previous code went through extra hoops to convert it to texture maps which is now done internally by the skin server)

File: src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java (hunk 1 above)
Comment (Sollace) [reply src/hdskins/java/com/voxelmodpack/hdskins/skins/NetClient.java]: Reset method was added with the eventual objective of being able to re-use the client based on what you were doing with clients in Valhalla.
